### PR TITLE
metrics: add instance var and improve conccurrency metrics

### LINKED ISSUE #7583: Use instance label wisely to optimize the presentation of Grafana
## Enhancement Task
After the microservices were implemented, the instance variable was removed from Grafana so that metrics for different instances could be displayed on the same page.
But this brings some issues:
1. The pd server does not reset metrics immediately after panic occurs. As a result, the monitored items using the sum function become larger such as #7506.
2. We support gRPC forwarding, which also causes the QPS we see to be higher than the actual one.
3. We will support the follower handle of gRPC and HTTP. We need to monitor QPS according to the instance.

### PR DESCRIPTION
**Pull request by CabinfeverB**
### What problem does this PR solve?

Issue Number: close #7583

### What is changed and how does it work?
add isntance var
add instance check for gRPC and HTTP
use `max_over_time` to  keep conccurrency metrics same with qps

```commit-message
```

### Check List

Tests

- Manual test (add detailed scripts or steps below)
image
image
image
image

conccurrency
image

Code changes

- Has the configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

File: metrics/grafana/pd.json [hunk 1]
```diff
@@ -132,7 +132,7 @@
       "tableColumn": "",
       "targets": [
         {
-          "expr": "sum(pd_cluster_status{k8s_cluster=\"$k8s_cluster\", tidb_cluster=\"$tidb_cluster\", type=\"storage_capacity\"})",
+          "expr": "max(pd_cluster_status{k8s_cluster=\"$k8s_cluster\", tidb_cluster=\"$tidb_cluster\", type=\"storage_capacity\"})",
```
Comment (rleungx) [inline metrics/grafana/pd.json:135]: If the original leader panics, and then offline a store, will the monitor always show the original value?

Comment (CabinfeverB) [reply metrics/grafana/pd.json]: Got it. This PR will not close #7506

**Review by rleungx — APPROVED** [review]

**Comment by CabinfeverB** [comment]
cc @nolouch @HuSharp

**Review by HuSharp — APPROVED** [review]

**Pull request merged**